### PR TITLE
Migrate RCStoreKitWrapper

### PR DESCRIPTION
**Pull request by aboedo**
Still a WIP, leaving as draft. 

Addresses #566

### CODE REVIEW

File: PurchasesCoreSwiftTests/Purchasing/StoreKitWrapperTests.swift [hunk 1]
```diff
@@ -146,7 +148,7 @@ class StoreKitWrapperTests: XCTestCase, RCStoreKitWrapperDelegate {
 
         wrapper?.paymentQueue(paymentQueue, updatedTransactions: [transaction])
 
-        wrapper?.finish(transaction)
+        wrapper?.finishTransaction(transaction)
```
Comment (aboedo) [inline PurchasesCoreSwiftTests/Purchasing/StoreKitWrapperTests.swift:151]: this is a slight naming change, but it more closely resembles the swift interface for [this method](https://developer.apple.com/documentation/storekit/skpaymentqueue)

File: PurchasesCoreSwiftTests/Purchasing/StoreKitWrapperTests.swift [hunk 2]
```diff
@@ -220,7 +220,7 @@ class StoreKitWrapperTests: XCTestCase, RCStoreKitWrapperDelegate {
 
             let mockProduct = MockSKProduct(mockProductIdentifier: productId)
             let mockDiscount = MockPaymentDiscount(mockIdentifier: discountId)
-            let payment = wrapper.payment(with: mockProduct, discount: mockDiscount)
+            let payment = wrapper.payment(withProduct: mockProduct, discount: mockDiscount)
```
Comment (aboedo) [inline PurchasesCoreSwiftTests/Purchasing/StoreKitWrapperTests.swift:225]: slight naming change for swift only. I can use `@objc` to keep the old interface. This one doesn't have a native `StoreKit` equivalent, so we should do whatever we like the most. 
I kinda like `withProduct`, but don't have a strong preference.

File: PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift [hunk 3]
```diff
@@ -25,8 +25,6 @@ import Foundation
     @objc public var paymentqueue_revoked_entitlements_for_product_identifiers: String { "PaymentQueue " +
         "didRevokeEntitlementsForProductIdentifiers: %@" }
     @objc public var paymentqueue_updatedtransaction: String { "PaymentQueue updatedTransaction: %@ %@ (%@) %@ - %d" }
-    @objc public var presenting_code_redemption_sheet_unavailable: String { "Attempted to present code redemption " +
-        "sheet, but it's not available on this device." }
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift:29]: this wasn't really necessary - we were using it on a method that was marking availability anyway

File: PurchasesCoreSwift/Purchasing/StoreKitWrapper.swift [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        paymentQueue.finishTransaction(transaction)
+    }
+
+    @available(iOS 14.0, *)
+    @available(macOS, unavailable)
+    @available(tvOS, unavailable)
+    @available(watchOS, unavailable)
+    @objc public func presentCodeRedemptionSheet() {
+        paymentQueue.presentCodeRedemptionSheet()
+    }
```
Comment (aboedo) [inline PurchasesCoreSwift/Purchasing/StoreKitWrapper.swift:76]: we used to have a check here to verify the OS version, but the `@available` check was already enough

File: PurchasesCoreSwiftTests/Purchasing/StoreKitWrapperTests.swift (hunk 2 above)
Comment (taquitos) [reply PurchasesCoreSwiftTests/Purchasing/StoreKitWrapperTests.swift:225]: I'm not opposed to this change, I think it's better. Now is the time to do updates like this IMO.

File: PurchasesCoreSwift/Purchasing/StoreKitWrapper.swift (hunk 4 above)
Comment (taquitos) [reply PurchasesCoreSwift/Purchasing/StoreKitWrapper.swift:76]: 🎉

File: Purchases/Public/RCPurchases.m [hunk 5]
```diff
@@ -1164,6 +1165,7 @@ - (void)                   storeKitWrapper:(RCStoreKitWrapper *)storeKitWrapper
     }];
 }
 
+
```
Comment (taquitos) [inline Purchases/Public/RCPurchases.m:1168]: extra line break

File: PurchasesCoreSwift/Purchasing/StoreKitWrapper.swift [hunk 6]
```diff
@@ -0,0 +1,147 @@
+//  RCStoreKitWrapper.m
+//  Purchases
+//
+//  Created by RevenueCat.
+//  Copyright © 2019 RevenueCat. All rights reserved.
```
Comment (taquitos) [inline PurchasesCoreSwift/Purchasing/StoreKitWrapper.swift:5]: 2019 👀

File: PurchasesCoreSwift/Purchasing/StoreKitWrapper.swift [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                paymentQueue.remove(self)
+            }
+        }
+    }
+
+    private var paymentQueue: SKPaymentQueue
+
+    @objc public init(paymentQueue: SKPaymentQueue) {
+        self.paymentQueue = paymentQueue
+        super.init()
```
Comment (taquitos) [inline PurchasesCoreSwift/Purchasing/StoreKitWrapper.swift:46]: `super.init()` not required when superclass is `boring` (your super class has a single designated initializer with a zero-argument init)

**Review by taquitos — APPROVED** [review]
Tiny nit about not needing to call `super.init()` in certain cases, but otherwise :shipit:

**Pull request merged**